pkg/common-controller/snapshot_controller_base.go

```diff
@@ -215,8 +215,11 @@ func (ctrl *csiSnapshotCommonController) snapshotWorker() {
 			// The volume snapshot still exists in informer cache, the event must have
 			// been add/update/sync
 			newSnapshot, err := ctrl.checkAndUpdateSnapshotClass(snapshot)
-			if err == nil {
-				klog.V(5).Infof("passed checkAndUpdateSnapshotClass for snapshot %q", key)
+			if err == nil || (newSnapshot.ObjectMeta.DeletionTimestamp != nil && errors.IsNotFound(err)) {
+				// If the VolumeSnapshotClass is not found, we still need to process an update
+				// so that syncSnapshot can delete the snapshot, should it still exist in the
+				// cluster after it's been removed from the informer cache
+				klog.V(5).Infof("passed checkAndUpdateSnapshotClass for snapshot %q, or snapshot needs to be deleted and snapshotClass doesn't exist", key)
```

Comment:
Suggest change to the following:
updating snapshot %q; snapshotClass may have already been removed

Reply:
This has been changed to use the suggested message.